Database/CleanArchive.py

```diff
@@ -88,12 +96,69 @@ def clean(all=True):
         transaction = {
             '$pull': {'snippets': {'transcript': ''}}
         }
+        if duplicate < 1.0:
+            print('Duplicate found')
+            set_fields['duplicate'] = True
+            duplicates['metadata.Subtitle'] += 1
+        else:
+            set_fields['duplicate'] = False
         if errors:
             set_fields['errors'] = errors
         else:
             transaction['$unset'] = {'errors': 1}
         if set_fields:  # $set cannot be empty.
             transaction['$set'] = set_fields
+
+
+
         db.ArchiveIndex.update_one({'_id': episode['_id']}, transaction)
     print('Updates:', updates)
     print('Failures:', failures)
+    print('Duplicates', duplicates)
+
+
+def checkDuplicate(epi):
+    upperBound = db.ArchiveIndex.find({'identifier': epi['identifier']})[0]['date']
+    dateTimeObj = datetime.datetime.strptime(upperBound, '%Y-%m-%dT%H:%M:%SZ')
+    lowerBound = str(dateTimeObj - dateTimeObj - datetime.timedelta(days=1))
+    fields = db.ArchiveIndex.find({'date': {'$lte': upperBound, '$gte': lowerBound}}, {'snippets': 1})
```

Comment:
Instead of searching for documents with snippets, just search the CleanedIndex instead of ArchiveIndex. If you haven't created the views yet, you can create them in the manage_db program.
Also, I don't like the name fields for the episodes to compare. compareEpisodes would be better.

Reply:
Small issue. If using cleanedIndex while ADDING errors to potential duplicates, it will NOT consider shows that are added as duplicated while iterating through this loop, so maybe your method is better.

Reply:
snippets is the projection not query🤦. Yeah this seems fine.
If anything, just make sure the Datetime_UTC field exists (and update your date to pull from the right field). This way it will only select from downloaded and cleaned episodes, reducing errors in the try/except loop.